File: Sber_backend_java/src/main/java/com/example/sber/model/dto/response/PrivilegeDto.java
package com.example.sber.model.dto.response;

import com.example.sber.model.enums.CurrentLevel;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PrivilegeDto {
    private String name;
    private String description;
    private int effectMoney;
    private boolean isActive;
    private CurrentLevel unlockedLevel;

    public PrivilegeDto (String name, String description, int effectMoney, boolean isActive, CurrentLevel unlockedLevel) {
        this.name = name;
        this.description = description;
        this.effectMoney = effectMoney;
        this.isActive = isActive;
        this.unlockedLevel = unlockedLevel;
    }

    public PrivilegeDto () {}
}

File: Sber_backend_kotlin/app/src/main/java/com/example/myapplication/data/model/PrivilegeDto.kt
package com.example.myapplication.data.model

import com.google.gson.annotations.SerializedName

data class PrivilegeDto(
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String,
    @SerializedName("effectMoney") val effectMoney: Int,
    @SerializedName("active") val isActive: Boolean,
    @SerializedName("unlockedLevel") val unlockedLevel: String?
)
